test/decl/nested/protocol_experimental.swift

```diff
@@ -0,0 +1,261 @@
+// RUN: %target-typecheck-verify-swift -parse-as-library -enable-experimental-feature NestedProtocols
```

Comment:
Please also add some IRGen tests to cover emission of protocol descriptors, and execution tests. In particular, I'm curious if a dynamic cast to a conditional conformance involving one of these things will work immediately, or require runtime changes:

```swift
enum E {
  protocol P {}
}

struct S<T> {}
extension S: E.P where T: E.P {}
extension Int: E.P {}

let x: Any = S<Int>()
print(x as? any E.P)
```